core/src/utils/mod.rs

```diff
@@ -29,13 +29,13 @@ pub const fn indices_arr<const N: usize>() -> [usize; N] {
     indices_arr
 }
 
-pub fn pad_to_power_of_two<const N: usize, T: Clone + Default>(values: &mut Vec<T>) {
-    debug_assert!(values.len() % N == 0);
-    let mut n_real_rows = values.len() / N;
+pub fn pad_to_power_of_two<T: Clone + Default>(num_columns: usize, values: &mut Vec<T>) {
```

Comment:
Is there a reason this needed to change? it seems like we are not using the fact it's a compile time constant, is that true?

Reply:
Actually, looks like I don't need to change it by using a const function to determine the poseidon2 width.  Will revert this.

Reply:
I actually now remember why I had to make this change.

So a const func that uses the struct's DEGREE generic wouldn't work.  The reason is that a const variable (event if declared within a function of the struct) can't access that generic.

`In Rust, const and static items are considered separate, standalone items, and thus they cannot capture or use generic parameters from their enclosing scopes directly. This limitation also applies to nested functions that attempt to use generics declared in an outer scope.`

E.g. these (illustrative) lines of code in `generate_trace` wouldn't work:

```
        const WIDTH: usize = if DEGREE > 3 { 16 } else { 4 };
        pad_to_power_of_two::<WIDTH, F>(&mut trace.values);
```

So that means that the poseidon2's chip's calculated width can't be a const, and can't be used as a const generic for `pad_to_power_of_two`.